### PR TITLE
Implementation of view model and Vuex in the shipping step

### PR DESCRIPTION
**Pull request by szafran89**

### CODE REVIEW

File: view/frontend/web/js/components/steps/BaseShippingStep.vue [hunk 1]
```diff
@@ -124,219 +240,30 @@ export default {
...
+    onCountryChange() {
+      this.regions = this.$store.getters.regionsByCountryId(this.countryId);
+      this.$store.dispatch('updateShippingMethods', this.countryId)
     },
-    returnError(element, cssClass, text) {
-      // Initial validation in future
+    onFormSubmit() {
+      // under this form data
+      console.log(this)
+      //TODO: save form data to store
```
Comment (kjugi) [inline view/frontend/web/js/components/steps/BaseShippingStep.vue]: Do you have an idea how to do this? I was thinking about it and not pretty sure I will have right

File: view/frontend/web/js/components/steps/BaseShippingStep.vue [hunk 2]
```diff
@@ -106,16 +207,31 @@ export default {
...
   data() {
     return {
-      baseUrl            : baseUrl,
-      config             : this.$store.state.config,
       isRegionIdHidden   : false,
       regionList         : regionList,
       shippingAddress    : shippingAddress,
       shippingInformation: this.$store.state.shippingInformation,
-      totals             : this.$store.state.totals
+      totals             : this.$store.state.totals,
```
Comment (kjugi) [inline view/frontend/web/js/components/steps/BaseShippingStep.vue]: can I add it as computed too?

File: view/frontend/web/js/components/steps/BaseShippingStep.vue [hunk 3]
```diff
@@ -106,16 +207,31 @@ export default {
...
+      postcode: '',
+      regionId: '',
+      company: '',
+      countries,
+      shippingMethod: '',
+      regions: []
     };
   },
   computed: {
+    baseUrl() {
```
Comment (kjugi) [inline view/frontend/web/js/components/steps/BaseShippingStep.vue]: Why you are adding here computed with baseUrl when we don't use it anywhere? Do you have a plan?

File: view/frontend/web/js/components/steps/BaseShippingStep.vue [hunk 4]
```diff
@@ -7,94 +7,195 @@
...
-  &--hidden {
-    display: none;
-  }
-}
-</style>
-
 <script>
 import BaseButton from '../BaseButton.vue';
 import BaseCheckbox from '../BaseCheckbox.vue';
 import BaseInput from '../BaseInput.vue';
```
Comment (kjugi) [inline view/frontend/web/js/components/steps/BaseShippingStep.vue]: we don't use few of this imports, is it necessary here?

File: view/frontend/web/js/components/steps/BaseShippingStep.vue (hunk 2 above)
Comment (szafran89) [reply view/frontend/web/js/components/steps/BaseShippingStep.vue]: We don't need it at all. Removed.

File: view/frontend/web/js/components/steps/BaseShippingStep.vue (hunk 1 above)
Comment (szafran89) [reply view/frontend/web/js/components/steps/BaseShippingStep.vue]: Sure, I have an idea :D

**Pull request merged**